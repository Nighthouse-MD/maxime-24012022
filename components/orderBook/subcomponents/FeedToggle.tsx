import { memo } from 'react';
import { Button } from 'react-bootstrap';
import styles from '../OrderBook.module.scss'

const FeedToggle = ({ handleFeedToggle }) => <Button
    id="toggleFeed"
    name="toggleFeed"
    className={styles.toggleButton}
    onClick={handleFeedToggle}
    size="sm"
>
    Toggle Feed
</Button>;

export default memo(FeedToggle);