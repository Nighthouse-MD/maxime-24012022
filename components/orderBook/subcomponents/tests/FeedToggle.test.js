import FeedToggle from '../FeedToggle';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent, getByText } from '@testing-library/react';

describe('FeedToggle', () => {
    it('shows the button', () => {
        const handleFeedToggleMock = jest.fn();
        render(<FeedToggle handleFeedToggle={handleFeedToggleMock} />);

        expect(screen.queryByText("Toggle Feed")).toBeInTheDocument();
    });

    it('fires the handler when clicking the button', () => {
        const handleFeedToggleMock = jest.fn();
        const { container } = render(<FeedToggle handleFeedToggle={handleFeedToggleMock} />);

        fireEvent(
            getByText(container, "Toggle Feed"),
            new MouseEvent("click", {
                bubbles: true,
                cancelable: true,
            }),
        );

        expect(handleFeedToggleMock).toHaveBeenCalledTimes(1);
    });

    it('renders correctly', () => {
        const handleFeedToggleMock = jest.fn();
        const tree = renderer
            .create(<FeedToggle handleFeedToggle={handleFeedToggleMock} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});