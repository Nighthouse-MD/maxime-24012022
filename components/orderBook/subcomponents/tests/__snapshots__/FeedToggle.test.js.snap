// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedToggle renders correctly 1`] = `
<button
  className="toggleButton btn btn-primary btn-sm"
  disabled={false}
  id="toggleFeed"
  name="toggleFeed"
  onClick={[MockFunction]}
  type="button"
>
  Toggle Feed
</button>
`;
